### PR TITLE
For #25374: Skip action validation for control messages

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt [hunk 1]
```diff
@@ -137,20 +137,26 @@ class NimbusMessagingStorage(
...
-            val safeAction = nimbusActions[unsafeAction]
-            if (safeAction.isNullOrBlank() || safeAction.isEmpty()) {
-                if (!malFormedMap.containsKey(unsafeAction)) {
-                    reportMalformedMessage(messageId)
+
+        return when {
+            unsafeAction.startsWith("http") -> {
+                unsafeAction
+            }
+            isControl -> ""
```
Comment (jhugman) [inline app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt:148]: Idle nit: is there anything more useful than the empty string you could use? e.g. for `logger.info` purposes, using `NOOP` or `PLACEBO` would more clearly signify a control message.

I don't know the answer, so feel free to disregard.

File: app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt [hunk 2]
```diff
@@ -137,20 +137,26 @@ class NimbusMessagingStorage(
...
+                if (safeAction.isNullOrBlank() || safeAction.isEmpty()) {
+                    if (!malFormedMap.containsKey(unsafeAction)) {
+                        reportMalformedMessage(messageId)
+                    }
+                    malFormedMap[unsafeAction] = messageId
+                    return null
                 }
-                malFormedMap[unsafeAction] = messageId
-                return null
+                safeAction
```
Comment (jhugman) [inline app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt:158]: Nice refactor.

**Review by jhugman — APPROVED** [review]
This is good. Thank you for the fast turn around.

File: app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt:148]: 👍🏽 
I'll will go with "CONTROL_ACTION"

**Comment by Amejia481** [comment]
@mergify  backport releases_v101.0.0

**Comment by Amejia481** [comment]
@Mergifyio backport releases_v101.0.0

**Pull request merged**